Automattic/sensei
# Create quiz progress on quiz answers submit

Resolves #7050

We create the quiz progress when start the lesson. When the user submits their answers we implicitly start the lesson.
Here we make sure we have a quiz progress or try to create one.
We don't check the feature flag as it is safe to create quiz progress for comments-based version: it is the same lesson progress there.

## Proposed Changes
* Create quiz progress on lesson progress creation if there is a quiz related to the lesson.

## Testing Instructions
1. Enable `tables_based_progress`.
2. Start a lesson (or go directly to a quiz and submit your answers).
3. Check sensei_lms_progress has quiz progress for the quiz. In case it was created on the lesson start `in-progress` status expected, on the quiz answers submit depends on quiz settings: could be ungraded, passed, failed, graded.


## Pre-Merge Checklist
